apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,150 @@
+import * as vscode from "vscode";
+import { WorkspaceConfiguration } from "vscode";
+
+const typeGenPatternKey = "*.ts";
+const typeGenPattern = "${capture}.typegen.ts";
+
+// Checks if the user has our typegen pattern set in their workspace config
+const hasTypeGenPattern = (fileNestingPatterns: object) =>
+  Object.values(fileNestingPatterns).some((pattern: string) =>
+    pattern
+      .split(",")
+      .map((part) => part.trim())
+      .some((part) => part === typeGenPattern)
+  );
+
+const enableTypeGenNesting = (
+  fileNestingConfig: WorkspaceConfiguration,
+  fileNestingPatterns: object
+) => {
+  // If the user chooses to enable file nesting for typegen files we need to make sure that VSCode's file nesting is also enabled
+  fileNestingConfig.update("enabled", true, true);
+
+  // Check if the user has a pattern defined for ts files
+  const existingTsPattern = fileNestingPatterns[typeGenPatternKey];
+
+  // If the user has defined a pattern for ts files we add our typegen pattern to it, otherwise we create a new pattern
+  const updatedPattern = existingTsPattern
+    ? `${existingTsPattern}, ${typeGenPattern}`
+    : typeGenPattern;
+
+  // Update file nesting patterns with all existing patterns and our updated pattern
+  fileNestingConfig.update(
+    "patterns",
+    {
+      ...fileNestingPatterns, // Reuse existing patterns
+      [typeGenPatternKey]: updatedPattern, // Add our updated pattern
+    },
+    true
+  );
+};
+
+const disableTypeGenNesting = (
+  xstateConfig: WorkspaceConfiguration,
+  fileNestingConfig: WorkspaceConfiguration,
+  fileNestingPatterns: object
+) => {
+  xstateConfig.update("nestTypegenFiles", false, true);
+
+  // Check if the user has a pattern defined for ts files
+  const existingTsPattern = fileNestingPatterns[typeGenPatternKey];
+
+  // If there's no patterns defined for ts files we don't need to do anything
+  if (existingTsPattern === "") return;
+
+  // Remove our typegen pattern from the existing patterns
+  const updatedPattern = existingTsPattern
+    .split(",")
+    .map((part) => part.trim())
+    .filter((part) => part !== typeGenPattern)
+    .join(",");
```

Comment:
nit: you are removing whitespace from the user's config - ideally we should preserve the original formatting

Reply:
Good catch, I've updated this to `...join(", ")` to keep the default vscode formating. OK?

Reply:
It's probably better as it matches the default formatting now - but a nitpick would be that it still doesn't preserve the input formatting. A version that would do this could look smth like this:
```js
  const updatedPattern = existingTsPattern
    .split(",")
    .filter((part) => {
        const trimmed = part.trim();
        return trimmed !== typeGenPattern;
    })
    .join(",");
```